### PR TITLE
remove Editor from Seed and CICD projects

### PR DESCRIPTION
**Pull request by daniel-cit**
remove Editor role  from Seed and CICD projects in 0-bootrstrap

### CODE REVIEW

File: 0-bootstrap/modules/parent-iam-remove-role/variables.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "parent_type" {
+  description = "Type of the parent resource. valid values are `organization`, `folder`, and `project`."
+  type        = string
+
+  validation {
+    condition     = var.parent_type == "organization" || var.parent_type == "folder" || var.parent_type == "project"
```
Comment (bharathkkb) [inline 0-bootstrap/modules/parent-iam-remove-role/variables.tf:22]: nit: might be simpler
```suggestion
    condition     = contains(["organization", "folder", "project"], var.parent_type)
```

File: 0-bootstrap/modules/parent-iam-remove-role/versions.tf [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      source  = "hashicorp/google"
+      version = ">= 3.77"
+    }
+    google-beta = {
+      source  = "hashicorp/google-beta"
+      version = ">= 3.77"
+    }
+    random = {
+      source = "hashicorp/random"
+    }
```
Comment (bharathkkb) [inline 0-bootstrap/modules/parent-iam-remove-role/versions.tf:30]: Doesn't look like we use both of these?

File: 0-bootstrap/sa.tf [hunk 3]
```diff
@@ -175,6 +175,31 @@ module "cicd_project_iam_member" {
   roles       = each.value
 }
 
+module "seed_project_remove_editor" {
+  source = "./modules/parent-iam-remove-role"
+
+  parent_type = "project"
+  parent_id   = module.seed_bootstrap.seed_project_id
```
Comment (bharathkkb) [inline 0-bootstrap/sa.tf:182]: nit: Does for_each throw an error due to computed project_id?

File: 0-bootstrap/sa.tf [hunk 4]
```diff
@@ -175,6 +175,31 @@ module "cicd_project_iam_member" {
   roles       = each.value
 }
 
+module "seed_project_remove_editor" {
```
Comment (bharathkkb) [inline 0-bootstrap/sa.tf:178]: It would be good to add a comment as a rationale.

**Review by bharathkkb — COMMENTED** [review]
Thanks @daniel-cit

File: 0-bootstrap/sa.tf (hunk 3 above)
Comment (daniel-cit) [reply 0-bootstrap/sa.tf]: we can use a `for_each` if we use the map option.
Good Catch.

File: 0-bootstrap/sa.tf (hunk 4 above)
Comment (daniel-cit) [reply 0-bootstrap/sa.tf]: Comment added

File: 0-bootstrap/modules/parent-iam-remove-role/variables.tf (hunk 1 above)
Comment (daniel-cit) [reply 0-bootstrap/modules/parent-iam-remove-role/variables.tf]: Fixed. 
Thanks for the suggestion.

File: 0-bootstrap/sa.tf [hunk 5]
```diff
@@ -176,6 +181,25 @@ module "cicd_project_iam_member" {
   roles       = each.value
 }
 
+// In the bootstrap projects is created, the Compute Engine
```
Comment (bharathkkb) [inline 0-bootstrap/sa.tf:184]:
```suggestion
// When the bootstrap projects are created, the Compute Engine
```

File: 0-bootstrap/sa.tf [hunk 6]
```diff
@@ -176,6 +181,25 @@ module "cicd_project_iam_member" {
...
+// default service account is disabled but it still has the Editor
+// role associated with the service account. When the projects are created
+// the Compute Engine default service account is the only Editor.
```
Comment (bharathkkb) [inline 0-bootstrap/sa.tf:187]:
```suggestion
// role associated with the service account. This default SA is the
// only member with the editor role.
```

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**